engines/bastion_katello/app/assets/javascripts/bastion_katello/repositories/details/repository-details-manage-ostree-branches.controller.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @ngdoc object
+ * @name  Bastion.repositories.controller:RepositoryManageContentController
+ *
+ * @requires $scope
+ * @requries $q
+ * @requires translate
+ * @requires Repository
+ *
+ * @description
+ *   Provides the functionality for the repository details pane.
+ */
+angular.module('Bastion.repositories').controller('RepositoryManageOstreeBranchesController',
+    ['$scope', '$q', 'translate', 'Repository',
+    function ($scope, $q, translate, Repository) {
+
+        $scope.wrap = function (repository) {
+            var branches = repository["ostree_branches"];
+            if (angular.isUndefined(branches)) {
+                branches = [];
+            }
+            $scope.branchObjects = _.map(branches, function (branch) {
+                return {name: branch};
+            } );
+        };
+
+
+        $scope.unwrap = function () {
+            return _.map($scope.branchObjects, function (branch) {
+                return branch.name;
+            } );
+        };
+
+        $scope.resetBranch = function () {
+            $scope.branch = {
+                editMode: false,
+                working: false
+            };
+        };
+
+        $scope.resetBranch();
+        $scope.repository = Repository.get({id: $scope.$stateParams.repositoryId});
+
+        $scope.repository.$promise.then($scope.wrap);
+
+        $scope.removeBranches = function () {
+            $scope.branchObjects = _.reject($scope.branchObjects, function (branch) {
+                return branch.selected;
+            });
+            $scope.saveBranches();
+        };
+
+        $scope.backupPrevious = function (branch) {
+            angular.copy(branch, branch.previous);
+        };
+
+        $scope.restorePrevious = function (branch) {
+            angular.copy(branch.previous, branch);
+            branch.previous = {};
+        };
+
+        $scope.saveBranches = function () {
+            var deferred = $q.defer();
+            $scope.repository["ostree_branches"] = $scope.unwrap();
+
+            $scope.repository.$update(function (repository) {
+                deferred.resolve(repository);
+                $scope.successMessages = [translate('Repository Branches Updated.')];
+                $scope.resetBranch();
+                $scope.wrap(repository);
+            }, function (response) {
+                deferred.reject(response);
+                _.each(response.data.errors, function (errorMessage) {
+                    $scope.errorMessages.push(translate("An error occurred saving the Repository: ") + errorMessage);
+                });
+            });
+
+            return deferred.promise;
```

Comment:
Since $scope.repository.$update should return a promise itself, why create your own?